### PR TITLE
Add landing page hero

### PR DESCRIPTION
**Pull request by oryxwhite**
This commit adds a hero section to the landing page.  The text content is pulled from the API.  The header and modal component folders were also moved to the new landing page folder.

### CODE REVIEW

**Review by mentor-tara — APPROVED** [review]
Go go go :smile:[](prolog-16)

File: pages/index.jsx [hunk 1]
```diff
@@ -1,7 +1,8 @@
 // import { Routes } from "@config/routes";
 import styles from "./index.module.scss";
-import { Header } from "../features/layout/header/index";
-import { Modal } from "../features/layout/modal/index";
+import { Header } from "../features/landing/components/header";
+import { Modal } from "../features/landing/components/modal";
+import { Hero } from "../features/landing/components/hero";
 import { useState } from "react";
 
 const IssuesPage = () => {
```
Comment (jkettmann) [inline pages/index.jsx:8]: This is a good chances to try out Static Site Generation (SSG) with Next.js using the `getStaticProps` function. This way the whole page is pre-rendered during build time. When a user visits the page they receive a ready-made HTML file which is a lot faster than first serving the skeleton of this page and then requesting the data on the client. 

I'd recommend checking out the Next.js docs about SSG and the `getStaticProps` function

File: features/landing/components/hero/hero.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
+import styles from "./hero.module.scss";
+import { useGetContent } from "../../api/use-get-content";
+
+export function Hero() {
+  const { data } = useGetContent();
```
Comment (jkettmann) [inline features/landing/components/hero/hero.tsx:5]: Apart from the comment I left on the index page I doesn't seem appropriate to fetch the content inside the Hero component. In fact, the API might not even return a hero section. I'd highly recommend checking out the video solution for this tasks. There are some nuggets in there 😄

File: api/landing.types.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+  title: string;
+  subtitle?: string;
+  image?: {
+    src: string;
+    width: number;
+    height: number;
+  };
+  companies?: Company[];
+  testimonials?: Testimonial[];
+};
```
Comment (jkettmann) [inline api/landing.types.ts:31]: This is actually 3 different section types, right? There's an alternative to this approach of mixing the types that also requires you to use optional fields:

Union types. So basically you define three separate types.

```
type HeroSection = {
sectionType: 'hero-section'; // or whatever the type was :)
  theme: string;
  title: string;
  subtitle: string;
  image: {
    src: string;
    width: number;
    height: number;
  };
}

type TestimonialSection = {
  testimonials?: Testimonial[];
}

...
```

Then you create a union type from these

```
type Section = HeroSection | TestimonialSection | ...
```

Again, I recommend watching the video solution as this is the approach I'm using. Then you can create a Section component that renders the appropriate section based on the type.

**Pull request merged**